runtime/Java/src/org/antlr/v4/runtime/atn/ATNSerializer.java

```diff
@@ -597,4 +634,11 @@ private void serializeInt(IntegerList data, int value) {
 		data.add((char)value);
 		data.add((char)(value >> 16));
 	}
+
+	private int encodeInt(int v) {
+		if ( language.equals("Java") && v==-1 ) {
+			return 0xFFFF;
```

Comment:
It looks incorrect since all int values have direct mapping. For Java Int32 encoding should be implemented as well.

Reply:
Yeah,I will think about the Java target once I get everybody else pure int.   most of the targets seemed to work with 32-bit ATN state numbers, but the unit tests ran out of memory running the big lexer via CI. also some of the targets don't work and I wasn't sure why.  I was going to make a separate issue to collect all of the thirty two bit atn state stuff that can then work off of this completed PR when merged.